net.solarnetwork.node.weather.ibm.wc/src/net/solarnetwork/node/weather/ibm/wc/BasicWCClient.java

```diff
@@ -0,0 +1,282 @@
+	public static final String DEFAULT_HOURLY_FORECAST_URL = "v3/wx/forecast/hourly/%s";
+
+	/** The default value for the {@code apiKeyTemplate} property. */
+	public static final String DEFAULT_API_KEY_TEMPLATE = "apiKey=%s";
+
+	/** The default value for the {@code languageTemplate} property. */
+	public static final String DEFAULT_LANGUAGE_TEMPLATE = "language=en-US";
+
+	/** The default value for the {@code units} property. */
+	public static final String DEFAULT_UNITS = "units=e";
+
+	private String baseUrl;
+	private String dailyForecastUrl;
+	private String hourlyForecastUrl;
+	private String defaultFormat;
+	private String apiKeyTemplate;
+	private String locationTemplate;
+	private String languageTemplate;
+	private String units;
+	private Boolean test;
+
+	private ObjectMapper objectMapper;
+
+	public BasicWCClient() {
+		this.baseUrl = DEFAULT_BASE_URL;
+		this.dailyForecastUrl = DEFAULT_DAILY_FORECAST_URL;
+		this.hourlyForecastUrl = DEFAULT_HOURLY_FORECAST_URL;
+		this.defaultFormat = DEFAULT_FORMAT;
+		this.apiKeyTemplate = DEFAULT_API_KEY_TEMPLATE;
+		this.locationTemplate = DEFAULT_LOCATION_TEMPLATE;
+		this.languageTemplate = DEFAULT_LANGUAGE_TEMPLATE;
+		this.units = DEFAULT_UNITS;
+		this.test = false;
+	}
+
+	private String getURL(String requestURL, String location, String apiKey, String datumPeriod) {
+		if ( test ) {
+			return this.baseUrl + "/" + String.format(requestURL, datumPeriod);
+		}
+		return this.baseUrl + "/" + String.format(requestURL, datumPeriod) + "?"
+				+ String.format(this.locationTemplate, location) + "&" + this.languageTemplate + "&"
+				+ this.defaultFormat + "&" + this.units + "&"
+				+ String.format(this.apiKeyTemplate, apiKey);
+	}
+
+	/**
+	 * For the daily forecast, each part of the data (ie. max temp, min temp,
+	 * narrative) is an array, with each entry corresponding with a day of the
+	 * week. The more detailed features in "daypart" such as cloud cover are
+	 * split into day and night, meaning two entries for each day.
+	 */
+	private BigDecimal parseBigDecimal(JsonNode n) {
+		if ( n.isNull() ) {
+			return null;
+		}
+		return new BigDecimal(n.asLong());
+	}
+
+	private LocalTime parseTimeValue(JsonNode n) {
+		if ( n.isNull() ) {
+			return null;
+		}
+		return new LocalTime(n.asLong() * 1000);
+
+	}
+
+	private Integer parseInt(JsonNode n) {
+		if ( n.isNull() ) {
+			return null;
+		}
+		return n.asInt();
+	}
+
+	//TODO valid time UTC
+	@Override
+	public Collection<GeneralDayDatum> readDailyForecast(String locationIdentifier, String apiKey,
+			String datumPeriod) {
+
+		if ( locationIdentifier == null ) {
+			return null;
+		}
+		final List<GeneralDayDatum> result = new ArrayList<GeneralDayDatum>();
+		final String url = getURL(this.dailyForecastUrl, locationIdentifier, apiKey, datumPeriod);
+		JsonNode root;
+		try {
+			log.debug("opening IBM Weather API connection");
+			log.debug(url);
+			URLConnection conn = getURLConnection(url, HTTP_METHOD_GET);
```

Comment:
I'd recommend using `JsonHttpClientSupport` and enabling the `compress` property by default, so that compressed HTTP response payloads are supported.